code/obj/storage/gang_crate.dm

```diff
@@ -0,0 +1,1257 @@
+			lootMaster.add_random_loot(src, GANG_CRATE_GEAR, 3)
+			// fill the rest with whatever
+			lootMaster.fill_remaining(src, GIMMICK)
+			..()
+		unlocked
+			New()
+				..()
+				anchored = UNANCHORED
+				locked = FALSE
+				src.light = image('icons/obj/large_storage.dmi',"gangcratefulllight")
+				UpdateIcon()
+	guns_and_gear_visualized
+		New()
+			..()
+			src.light = image('icons/obj/large_storage.dmi',"gangcratefulllight")
+			UpdateIcon()
+			anchored = UNANCHORED
+			locked = FALSE
+			lootMaster =  new /datum/loot_generator(4,4)
+			src.open()
+			// 3 guns, ammo, 3 bits of gear
+			for (var/i=1 to 3)
+				lootMaster.add_random_loot(src, GANG_CRATE_GUN, 1)
+				sleep(1 SECOND)
+			for (var/i=1 to 3)
+				lootMaster.add_random_loot(src, GANG_CRATE_AMMO_LIMITED, 1)
+				sleep(1 SECOND)
+			for (var/i=1 to 3)
+				lootMaster.add_random_loot(src, GANG_CRATE_GEAR, 1)
+				sleep(1 SECOND)
+			// fill the rest with whatever
+			lootMaster.fill_remaining(src, GIMMICK)
+	shotguns
+		New()
+			lootMaster =  new /datum/loot_generator(4,3)
+			lootMaster.place_loot_instance(src, 1,3, new /obj/randomloot_spawner/long/striker, FALSE)
+			lootMaster.place_loot_instance(src, 1,2, new /obj/randomloot_spawner/long/striker, FALSE)
+			lootMaster.fill_remaining(src, GANG_CRATE_AMMO, 3)
+			..()
+		unlocked
+			New()
+				..()
+				anchored = UNANCHORED
+				locked = FALSE
+				src.light = image('icons/obj/large_storage.dmi',"gangcratefulllight")
+				UpdateIcon()
+	only_gimmicks
+		New()
+			lootMaster =  new /datum/loot_generator(4,3)
+			lootMaster.fill_remaining(src.loc, GIMMICK)
+			..()
+		unlocked
+			New()
+				..()
+				anchored = FALSE
+				locked = FALSE
+				src.light = image('icons/obj/large_storage.dmi',"gangcratefulllight")
+				UpdateIcon()
+	only_guns
+		New()
+			lootMaster =  new /datum/loot_generator(4,3)
+			lootMaster.fill_remaining(src, GANG_CRATE_GUN)
+			..()
+	only_gear
+		New()
+			lootMaster =  new /datum/loot_generator(4,3)
+			lootMaster.fill_remaining(src, GANG_CRATE_GEAR)
+			..()
+	gear_and_gimmicks
+		New()
+			lootMaster =  new /datum/loot_generator(4,3)
+			lootMaster.add_random_loot(src, GANG_CRATE_GEAR, 2)
+			lootMaster.fill_remaining(src, GIMMICK)
+			..()
+
+	disposing()
+		STOP_TRACKING_CAT(TR_CAT_GHOST_OBSERVABLES)
+		..()
+
+	New()
+		..()
+
+		START_TRACKING_CAT(TR_CAT_GHOST_OBSERVABLES)
+		src.light = image('icons/obj/large_storage.dmi',"gangcratefulllight")
+		SPAWN(GANG_CRATE_LOCK_TIME/3)
+			src.light = image('icons/obj/large_storage.dmi',"gangcratehalflight")
+			UpdateIcon()
+		SPAWN(2*GANG_CRATE_LOCK_TIME/3 )
+			src.light = image('icons/obj/large_storage.dmi',"gangcratelowlight")
+			UpdateIcon()
+		SPAWN((GANG_CRATE_LOCK_TIME - 15 SECONDS) )
+			src.light = image('icons/obj/large_storage.dmi',"gangcrateblinkinglight")
+			UpdateIcon()
+		SPAWN(GANG_CRATE_LOCK_TIME)
+			src.light = image('icons/obj/large_storage.dmi',"gangcratefulllight")
+			anchored = FALSE
+			UpdateIcon()
+
+	ex_act()
```

Comment:
Might also want to consider `emag_act` and `meteor_hit` as rare as those are going to be.

Reply:
makes sense due to the superclass. ill just return for now unless that's a dumb idea